nodejs/eks/cmd/provider/index.ts

```diff
@@ -13,27 +13,104 @@
 // limitations under the License.
 
 import * as pulumi from "@pulumi/pulumi";
-import { Cluster } from "../../cluster";
+import { clusterProviderFactory } from "./cluster";
+import { vpcCniProviderFactory } from "./cni";
+import { randomSuffixProviderFactory } from "./randomSuffix";
 
 class Provider implements pulumi.provider.Provider {
     readonly version = getVersion();
 
+    // A map of types to provider factories. Calling a factory may return a new instance each
+    // time or return the same provider instance.
+    private readonly typeToProviderFactoryMap: Record<string, () => pulumi.provider.Provider> = {
+        "eks:index:Cluster": clusterProviderFactory,
+        "eks:index:RandomSuffix": randomSuffixProviderFactory,
+        "eks:index:VpcCni": vpcCniProviderFactory,
+    };
+
+    check(urn: pulumi.URN, olds: any, news: any): Promise<pulumi.provider.CheckResult> {
+        const provider = this.getProviderForURN(urn);
+        if (!provider) { return unknownResourceRejectedPromise(urn); }
+        return provider.check
+            ? provider.check(urn, olds, news)
+            : Promise.resolve({ inputs: news, failures: [] });
+    }
+
+    diff(id: pulumi.ID, urn: pulumi.URN, olds: any, news: any): Promise<pulumi.provider.DiffResult> {
+        const provider = this.getProviderForURN(urn);
+        if (!provider) { return unknownResourceRejectedPromise(urn); }
+        return provider.diff
+            ? provider.diff(id, urn, olds, news)
+            : Promise.resolve({});
+    }
+
+    create(urn: pulumi.URN, inputs: any): Promise<pulumi.provider.CreateResult> {
+        const provider = this.getProviderForURN(urn);
+        return provider && provider.create
```

Comment:
Any reason not to use optional chaining here?
```suggestion
        return provider?.create
```